internal/server/server.go

```diff
@@ -339,12 +334,18 @@ func (s *Server) handleLogin() http.HandlerFunc {
 		}
 		session.Values["nonce"] = nonce
 
+		// generate a PKCE code verifier
+		codeVerifier := oauth2.GenerateVerifier()
+
+		// store code_verifier for token exchange
+		session.Values["code_verifier"] = codeVerifier
+
 		if err := session.Save(r, w); err != nil {
 			logAndError(w, http.StatusInternalServerError, err, "error saving session")
 			return
 		}
 
-		http.Redirect(w, r, s.oauth2Config.AuthCodeURL(randomStr, oidc.Nonce(nonce)), http.StatusSeeOther)
+		http.Redirect(w, r, s.oauth2Config.AuthCodeURL(state, oidc.Nonce(nonce), oauth2.S256ChallengeOption(codeVerifier)), http.StatusSeeOther)
```

Comment:
Enabling PKCE should be behind a feature flag as not all IDPs support PKCE or public clients.